zmq.hpp

```diff
@@ -2030,6 +2030,13 @@ enum class socket_type : int
     client = ZMQ_CLIENT,
     radio = ZMQ_RADIO,
     dish = ZMQ_DISH,
+    gather = ZMQ_GATHER,
```

Comment:
Should probably guard all with `#ifdef`s?

Reply:
They're all behind `#if defined(ZMQ_BUILD_DRAFT_API) && ZMQ_VERSION >= ZMQ_MAKE_VERSION(4, 2, 0)` - I verified the version for all.